vuepress/docs/next/tutorials/devops/extend-plugin-descriptor.md

```diff
@@ -0,0 +1,69 @@
+---
+sidebarDepth: 2
+---
+
+# Leverage Plugin Environment Variables 
+
+This tutorial describes how to use environment variables and secrets in the plugin descriptor  of an Entando Component Generator built project to customize microservices behavior. You can use the same docker image to run during development as in production by using the environment variables as a configuration mechanism for your pods or for passing sensitive information. A new validation step is embedded in the Entando process to manage possible incompatible properties.
+
+
+## Entando Bundle Plugin Descriptor
+Currently, environment variables or secrets are not required for the container plugin descriptor file. The plugin descriptor v4 on the Entando Platform adds support for inserting environment variables as allowed by the descriptor.yaml file in k8s. There are two ways to accomplish this. You can deploy a plugin and inject the variables by using a key-value or by reading the value with an existing secret. The bundle is able to receive the configuration values exposed as plain text on the resources/yamls or hidden in the secret. Secrets can be referenced by using a special syntax. This process does not work with Entando plugin descriptor v3 or lower.
+
+### Deploy an EntandoApp with a new secret constraints: ??? not sure what this step is
```

Comment:
The language is odd. It means that you add a new secret in the namespace. We'll need to find the command for that.  The --from-literal option is probably the simplest so we can provide a command for it - https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/. Something like 'kubectl create secret generic env-2-secret --from-literal=env-2-secret-key=env2value -n entando'

Reply:
secrets -> Secrets

should format v4 - quotes? or format "plugin descriptor v4" as a proper noun? same with the v3 reference

files are in backticks -- `descriptor.yaml`

"in the resources/YAMLS"

"...referenced by using..." -- can omit "by"

can omit "There are two ways to accomplish this." and add "either" to the next sentence: "You can deploy a plugin and inject the variables either by..."

"key-value pair" to use complete terms

"with an existing Secret" doesn't sound right... can be read as 'the value has an existing Secret' 
change to something like "...by using a key-value pair or an existing Secret to read the value."

"The bundle is able to receive the configuration values exposed as plain text on the resources/yamls or hidden in the secret. Secrets can be referenced by using a special syntax." -- the sentence "Secrets can be referenced..." doesn't intuitively follow the preceding sentence; is referencing a Secret the same as reading it? and what is the special syntax?